examples/daal4py/model_builders_xgboost_early_stop_batch.py

```diff
@@ -0,0 +1,119 @@
+#===============================================================================
+# Copyright 2014 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+# daal4py Gradient Bossting Classification model creation from XGBoost example
+
+import daal4py as d4p
+import xgboost as xgb
+import numpy as np
+import pandas as pd
+from scipy.sparse import csr_matrix
+from sklearn.datasets import make_classification
+from sklearn.model_selection import train_test_split
+
+
+def pd_read_csv(f, c=None, t=np.float64):
+    return pd.read_csv(f, usecols=c, delimiter=',', header=None, dtype=t)
+
+
+def main(readcsv=pd_read_csv, method='defaultDense'):
+    # Path to data
+    train_file = "./data/batch/df_classification_train.csv"
+    test_file = "./data/batch/df_classification_test.csv"
+
+    ''' Data reading
+    Note that we cannot use sparse format for class labels
+    if we are using sklearn-like interface
+    '''
+    X_train = readcsv(train_file, range(3), t=np.float32)
+    y_train = readcsv(train_file, range(3, 4), t=np.float32)
+    X_test = readcsv(test_file, range(3), t=np.float32)
+    y_test = readcsv(test_file, range(3, 4), t=np.float32)
+
+    # training parameters setting
+    params = {
+        'n_estimators': 200,
+        'max_bin': 256,
+        'scale_pos_weight': 2,
+        'lambda_l2': 1,
+        'alpha': 0.9,
+        'max_depth': 8,
+        'num_leaves': 2**8,
+        'verbosity': 0,
+        'objective': 'multi:softmax',
+        'learning_rate': 0.3,
+        'num_class': 5,
+        'early_stopping_rounds': 5
+    }
+
+    # Training
+    xgb_clf = xgb.XGBClassifier(**params)
+    xgb_clf.fit(X_train, y_train, eval_set=[(X_test, y_test)])
```

Comment:
Do we really want the validation set is printed to stdout at each boosting stage?

Reply:
You should provide eval_set to fit xgboost with early_stopping_rounds (this feature is used to avoid overfitting so dataset different from train should be provided)